cmd/manifest/manifest.go

```diff
@@ -4,83 +4,92 @@ import (
+		return err.Error()
+	}
 
-	// Define an array of images.
-	skupperImages := []SkupperImage{
-		{
-			Name:       images.GetRouterImageName(),
-			Repository: "https://github.com/skupperproject/skupper-router",
-		},
-		{
-			Name:       images.GetServiceControllerImageName(),
-			Repository: "https://github.com/skupperproject/skupper",
-		},
-		{
-			Name:       images.GetConfigSyncImageName(),
-			Repository: "https://github.com/skupperproject/skupper",
-		},
-		{
-			Name:       images.GetFlowCollectorImageName(),
-			Repository: "https://github.com/skupperproject/skupper",
-		},
-		{
-			Name: images.GetPrometheusServerImageName(),
-		},
+	// Get the image digest
+	digestCmd := exec.Command("docker", "inspect", "--format='{{index .RepoDigests 0}}'", imageName)
+	digestBytes, err := digestCmd.Output()
+	if err != nil {
+		fmt.Printf("Error getting image digest: %v", err)
+		return err.Error()
 	}
 
+	imageWithoutTag := strings.Split(imageName, ":")[0]
+
+	// Extract and print the digest
+	parsedDigest := strings.ReplaceAll(strings.ReplaceAll(string(digestBytes), "'", ""), "\n", "")
+	digest := strings.TrimPrefix(strings.Trim(parsedDigest, "'"), imageWithoutTag+"@")
+
+	return digest
+}
+
+func main() {
+
 	// Define a struct for the manifest file.
 	manifest := struct {
 		Images []SkupperImage `json:"images"`
 	}{
-		Images: skupperImages,
+		Images: []SkupperImage{
+			{
+				Name:       images.GetRouterImageName(),
+				SHA:        getSha(images.GetRouterImageName()),
+				Repository: "https://github.com/skupperproject/skupper-router",
+			},
+			{
+				Name:       images.GetServiceControllerImageName(),
+				SHA:        getSha(images.GetServiceControllerImageName()),
+				Repository: "https://github.com/skupperproject/skupper",
+			},
+			{
+				Name:       images.GetConfigSyncImageName(),
+				SHA:        getSha(images.GetConfigSyncImageName()),
+				Repository: "https://github.com/skupperproject/skupper",
+			},
+			{
+				Name:       images.GetFlowCollectorImageName(),
+				SHA:        getSha(images.GetFlowCollectorImageName()),
+				Repository: "https://github.com/skupperproject/skupper",
+			},
+			{
+				Name: images.GetPrometheusServerImageName(),
+				SHA:  getSha(images.GetPrometheusServerImageName()),
+			},
+		},
 	}
 
 	// Encode the manifest image list as JSON.
 	manifestListJSON, err := json.MarshalIndent(manifest, "", "   ")
 	if err != nil {
-		return fmt.Errorf("Error encoding manifest image list: %v\n", err)
+		panic(fmt.Errorf("Error encoding manifest image list: %v\n", err))
 
 	}
 
 	// Create a new file.
 	file, err := os.Create("manifest.json")
 	if err != nil {
-		fmt.Errorf("Error creating file: %v\n", err)
+		panic(fmt.Errorf("Error creating file: %v\n", err))
 	}
 
 	// Write the JSON data to the file.
 	_, err = file.Write(manifestListJSON)
 	if err != nil {
-		return fmt.Errorf("Error writing to file: %v\n", err)
-	}
-
-	return nil
-}
-
-func main() {
-	var command = &cobra.Command{
```

Comment:
Agree keeping the cobra might be more than needed, but a --help might be good and can highlight the dependency on docker to complete the manifest generation. We may find that we ask users to run this utility to gather remote registry details, etc.

Reply:
Updated in 05739d27b7002f390b76ebb82ac4d24b24ea714e